templates/careers/application/partial/_assessment.html

```diff
@@ -1,38 +1,83 @@
+{% set assessment_stage | trim %}
+  {% if 'stage_progress' in application and application['stage_progress']['early_stage'] %}
+    complete
+  {% elif 'stage_progress' in application and not application['stage_progress']['early_stage'] and application['stage_progress']['assessment'] %}
+    active
+  {% else %}
+    not-completed
+  {% endif %}
+{% endset %}
+
+{% set is_early_stage = "stage_progress" in application and application["stage_progress"]["assessment"] and application["stage_progress"]["early_stage"] %}
```

Comment:
Maybe for a future refactor: I think it would make sense to move this logic to the flask view, as you say. `is_early_stage` sound like a method on and `Application` object that doesn't exist now (it's just a `dict`)